docs/search/docsearch-scraper.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash -x
+# Copyright © SixtyFPS GmbH <info@slint.dev>
+# SPDX-License-Identifier: MIT
+
+build_docs=false
+
+while getopts a:bc:d:i:p:r:u: flag
+do
+    case "${flag}" in
+        a) api=${OPTARG};;
+        b) build_docs=true;;
+        c) config=${OPTARG};;
+        d) docs=${OPTARG};;
+        i) index=${OPTARG};;
+        p) port=${OPTARG};;
+        r) remote=${OPTARG};;
+        u) url=${OPTARG};;
+    esac
+done
+
+# Get the directory of the current script
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# Set defaults
+if [ -z "$api" ]; then
+  api=xyz
+fi
+if [ -z "$config" ]; then
+  config="$script_dir/scraper-config.json"
+fi
+if [ -z "$docs" ]; then
+  docs="$script_dir/../../target/slintdocs/html"
+fi
+if [ -z "$index" ]; then
+  index=local
+fi
+if [ -z "$port" ]; then
+  port=8108
+fi
+if [ -n "$remote" ]; then
+  host=$remote
+  server=$remote
+  protocol=https
+  port=443
+  hostport=$remote
+else
+  host=localhost
+  server=host.docker.internal
+  protocol=http
+  port=$port
+  hostport=$host:$port
+fi
+if [ -z "$url" ]; then
+  url=http://localhost:8000
+fi
+
+# echo "Script Directory: $script_dir";
+# echo "API Key: $api";
+# echo "Config: $config";
+# echo "Docs: $docs";
+# echo "Host: $host";
+# echo "Index: $index";
+# echo "Port: $port";
+# echo "Protocol: $protocol";
+# echo "Server: $server";
+# echo "Url: $url";
+# echo "HostPort: $hostport";
+
+# Update server and api in searchbox.html and build slint docs
+if $build_docs; then
+  searchbox_html="$script_dir/../reference/_templates/searchbox.html"
+  cp $searchbox_html temp_searchbox.html
+  sed -i "s/\$TYPESENSE_SEARCH_API_KEY/$api/g" $searchbox_html
+  sed -i "s/\$TYPESENSE_SERVER_PROTOCOL/$protocol/g" $searchbox_html
+  sed -i "s/\$TYPESENSE_INDEX_NAME/$index/g" $searchbox_html
+  sed -i "s/\$TYPESENSE_SERVER_PORT/$port/g" $searchbox_html
+  sed -i "s/\$TYPESENSE_SERVER_URL/$host/g" $searchbox_html
+  cargo xtask slintdocs --show-warnings
+fi
+
+# Start http server
+python3 -m http.server 80 -d $docs &
+
+# Update index name in config file
+cp $config temp_config.json
+config=temp_config.json
+sed -i "s/\$TYPESENSE_INDEX_NAME/$index/g" $config
+
+
+# Run docsearch-scraper
+docker run -i \
+  -e "TYPESENSE_API_KEY=$api" \
+  -e "TYPESENSE_HOST=$server" \
+  -e "TYPESENSE_PORT=$port" \
+  -e "TYPESENSE_PROTOCOL=$protocol" \
+  -e "CONFIG=$(cat $config | jq -r tostring)" \
+  typesense/docsearch-scraper:0.10.0 | tee temp_scraper_output.txt
+
+# Kill http server
+killall python
+killall Python
+
+# Retrieve the collection name
+pattern=$index'_[0-9]\+'
+collection_name=$(grep -o -m 1 $pattern temp_scraper_output.txt)
```

Comment:
This logic to determine the collection name doesn't work for me (causing the rest to fail). This is because the collection name output for me appears in stderr, not stdout. Proposed fix below.